storage/ndb/rest-server/data-access-rondb/src/db-operations/pk/common.cpp

```diff
@@ -348,40 +354,88 @@ RS_Status SetOperationPKCol(const NdbDictionary::Column *col, NdbOperation *oper
     [[fallthrough]];
   case NdbDictionary::Column::Longvarchar: {
     ///< Length bytes: 2, little-endian
-    const int len = request->PKValueLen(colIdx);
-    if (len > col->getLength()) {
+    const int data_len = request->PKValueLen(colIdx);
+    if (data_len > col->getLength()) {
       return RS_CLIENT_ERROR(
           std::string(ERROR_008) +
-          " Data len is greater than column length. Column: " + std::string(col->getName()));
+          " Data length is greater than column length. Column: " + std::string(col->getName()));
     }
     char *charStr;
     if (request->PKValueNDBStr(colIdx, col, &charStr) != 0) {
       return RS_CLIENT_ERROR(ERROR_019);
     }
-    if (operation->equal(request->PKName(colIdx), charStr, len) != 0) {
+    if (operation->equal(request->PKName(colIdx), charStr, data_len) != 0) {
       return RS_SERVER_ERROR(ERROR_023);
     }
     return RS_OK;
   }
   case NdbDictionary::Column::Binary: {
     ///< Len
-    // we get the data in base64
     const char *encodedStr = request->PKValueCStr(colIdx);
-    size_t decoded_size = boost::beast::detail::base64::decoded_size(request->PKValueLen(colIdx));
-    int maxlen          = std::max(col->getLength(), static_cast<int>(decoded_size));
+    int data_len = request->PKValueLen(colIdx);
 
-    char pk[maxlen];
-    for (int i = 0; i < col->getLength(); i++) {
-      pk[i] = 0;
+    /*
+      Binary data will be sent as base64 strings.
+      These use 4 ASCII characters to represent any 3 bytes.
+
+      This means that any data which is not a multiple of 3 bytes will
+      append the padding character "=" once or twice in the encoded string.
+
+      E.g. the string "a"
+        -> to byte array: [ 97 ]
+        -> to base64 string: "YQ=="
+      This uses "==" as padding, since we only have 1 byte to represent.
+
+      Since the function decoded_size() does not read the string, it is not aware
+      of whether it contains padding characters or not. PKValueLen() is also not
+      aware of this. Thereby, any "==" will be intepreted as part of the data
+      and therefore decoded_size() will always return a multiple of 3 bytes.
+
+      E.g. given base64 string "YQ==" (4 ASCII characters)
+        -> data_len == 4 bytes
+        -> decoded_size == 3 bytes (since base64 always uses 4 bytes to represent 3 bytes of data)
+    */
+    size_t decoded_size = boost::beast::detail::base64::decoded_size(data_len);
```

Comment:
this isn't very easy to understand. There is no need to put base64 encoding-specific logic here. 

remove `boost::beast::detail::base64::decoded_size(data_len);` and call `std::pair<std::size_t, std::size_t> ret = boost::beast::detail::base64::decode(pk, encodedStr, data_len);`  at the start and use the size returned by this function. it returns a tuple and you can use these two values for error checking.